test/integration/TokenRejectIntegrationTest.js

```diff
@@ -0,0 +1,834 @@
+            ).getReceipt(env.client);
+
+            const tokenBalanceQuery = await new AccountBalanceQuery()
+                .setAccountId(receiverId)
+                .execute(env.client);
+
+            const tokenBalance = tokenBalanceQuery.tokens.get(nftId);
+            const tokenBalance2 = tokenBalanceQuery.tokens.get(nftId2);
+
+            expect(tokenBalance).to.be.equal(null);
+            expect(tokenBalance2).to.be.equal(null);
+        });
+
+        it("should return tokens back to treasury receiverSigRequired is true", async function () {
+            this.timeout(1200000);
+
+            await new AccountUpdateTransaction()
+                .setAccountId(env.operatorId)
+                .setReceiverSignatureRequired(true)
+                .execute(env.client);
+
+            const nftId = new NftId(tokenId, 1);
+            const transferTransactionResponse = await new TransferTransaction()
+                .addNftTransfer(nftId, env.operatorId, receiverId)
+                .freezeWith(env.client)
+                .execute(env.client);
+
+            await transferTransactionResponse.getReceipt(env.client);
+
+            const tokenRejectResponse = await (
+                await new TokenRejectTransaction()
+                    .addNftId(nftId)
+                    .setOwnerId(receiverId)
+                    .freezeWith(env.client)
+                    .sign(receiverPrivateKey)
+            ).execute(env.client);
+
+            await tokenRejectResponse.getReceipt(env.client);
+
+            const tokenBalanceTreasuryQuery = await new AccountBalanceQuery()
+                .setAccountId(env.operatorId)
+                .execute(env.client);
+
+            const tokenBalanceTreasury = tokenBalanceTreasuryQuery.tokens
+                .get(tokenId)
+                .toInt();
+            expect(tokenBalanceTreasury).to.be.equal(1);
+
+            const tokenBalanceReceiverQuery = await new AccountBalanceQuery()
+                .setAccountId(receiverId)
+                .execute(env.client);
+
+            const tokenBalanceReceiver = tokenBalanceReceiverQuery.tokens
+                .get(tokenId)
+                .toInt();
+            expect(tokenBalanceReceiver).to.equal(0);
+        });
+
+        it("should return spender allowance to 0 after owner rejects NFT", async function () {
+            this.timeout(120000);
+
+            // create spender account
+            const spenderAccountPrivateKey = PrivateKey.generateED25519();
+            const spenderAccountTx = await new AccountCreateTransaction()
+                .setMaxAutomaticTokenAssociations(-1)
+                .setInitialBalance(new Hbar(10))
+                .setKey(spenderAccountPrivateKey)
+                .execute(env.client);
+
+            const { accountId: spenderAccountId } =
+                await spenderAccountTx.getReceipt(env.client);
+
+            // transfer nft to receiver
+            const nftId = new NftId(tokenId, 1);
```

Comment:
Isn't this already defined in the `beforeEach`?

Reply:
Valid here and think i spotted it in more places.

Reply:
It is not, but it can be moved there.